packages/schematics/angular/migrations/update-6/index.ts

```diff
@@ -176,23 +176,96 @@ function extractProjectsConfig(config: CliConfig): JsonObject {
       const outDir = app.outDir || defaults.outDir;
       const appRoot = app.root || defaults.appRoot;
 
-      const project: JsonObject = {
-        root: '',
-        projectType: 'application',
-        cli: {},
-        schematics: {},
-      };
+      function _mapAssets(asset: string | JsonObject) {
+        if (typeof asset === 'string') {
+          return { glob: asset, input: appRoot + '/', output: '/' + outDir + '/' };
+        } else {
+          return {
+            glob: asset.glob,
+            input: appRoot + '/' + asset.input,
+            output: outDir + '/' + asset.output,
+          };
+        }
+      }
+
+      function _buildConfigurations(): JsonObject {
+        const source = app.environmentSource;
+        const environments = app.environments;
+
+        if (!environments) {
+          return {};
+        }
+
+        return Object.keys(environments).reduce((acc, environment) => {
+          let isProduction = false;
+
+          const environmentContent = tree.read(app.root + '/' + environments[environment]);
+          if (environmentContent) {
+            isProduction = !!environmentContent.toString('utf-8')
+              // Allow for `production: true` or `production = true`. Best we can do to guess.
+              .match(/production['"]?\s*[:=]\s*true/);
+          }
 
-      const extraEntryMapper = (extraEntry: string | JsonObject) => {
+          acc[environment] = {
+            ...(isProduction
+              ? {
+                optimization: true,
+                outputHashing: 'all',
+                sourceMap: false,
+                extractCss: true,
+                namedChunks: false,
+                aot: true,
+                extractLicenses: true,
+                vendorChunk: false,
+                buildOptimizer: true,
+              }
+              : {}
+            ),
+            fileReplacements: [
+              { from: `${app.root}/${source}`, to: `${outDir}/${environments[environment]}` },
+            ],
+          };
+
+          return acc;
+        }, {} as JsonObject);
+      }
+
+      function _serveConfigurations(): JsonObject {
+        const environments = app.environments;
+
+        if (!environments) {
+          return {};
+        }
+
+        return Object.keys(environments).reduce((acc, environment) => {
+          acc[environment] = { browserTarget: `${name}:build:${environment}` };
+
+          return acc;
+        }, {} as JsonObject);
+      }
+
+      function _extraEntryMapper(extraEntry: string | JsonObject) {
         let entry: JsonObject;
         if (typeof extraEntry === 'string') {
-          entry = { input: extraEntry };
+          entry = { input: join(app.root as Path, extraEntry) };
         } else {
-          entry = extraEntry;
+          const input = join(app.root as Path, extraEntry.input as string || '');
+          const lazy = !!extraEntry.lazy;
+          entry = { input, lazy };
+
+          if (!extraEntry.output && !lazy) {
```

Comment:
`!lazy` -> `lazy`. It only needs to be inferred for lazy entries.

Reply:
Done.